src/main/scala/au/com/agiledigital/toolform/model/Project.scala

```diff
@@ -124,4 +118,18 @@ final case class Link(from: Reference, to: Reference) {
   * @param from the resource that supplies the volume.
   * @param to   the Project element that will be supplied the volume.
   */
-final case class Volume(from: Resource, to: Reference)
+final case class Volume(from: Reference, to: Reference) {
+  def resolve(project: Project): ResolvedLink = {
```

Comment:
Is it difficult to refactor this to return an Option[ResolvedLink]? Then your matching below can be rewritten as a for comprehension and we don't need to deal with throwing exceptions here

Reply:
I took inspiration for this resolve method from the `link` class above this `Volume` class. I think it was written in this way to be a reusable way to throw a detailed error if incorrect components or resources were referenced in the links/volumes section of the topology.conf. let me know if you still think they should be changed to Options[]. Resolve for a Link object is called in `InspectCommand.scala` and so this would have to be refactored too if they are changed.

This did, however, reminded me that I should be using the resolve method for volume topologies, so I included it in line 142 of `DeploymentWriter.scala`.